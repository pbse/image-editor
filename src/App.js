// Entrance File For This App

import React, { Component } from 'react';
import Dropzone from 'react-dropzone';
import Editor from '../src/Editor'
import './App.css';
import 'bootstrap/dist/css/bootstrap.css';

class App extends Component {
  
  constructor(props){
    super(props);

    this.state = {
      uploadedFile: null,
    };
  }

  /* Function to save Image
     Could write logic to 
     save it in server by 
     making a post call
  */ 
  
  onImageDrop(files) {
    this.setState({
      uploadedFile: files[0]
    });
  }

  render() {
    return (
      <div className="container">
      <div className="row">
      <form>
        <div className="FileUpload col-md-2">
          <Dropzone
            onDrop={this.onImageDrop.bind(this)}
            multiple={false}
            accept="image/*"
            style={{margin: '20px', width:'160px', height:'80px', border: '1px solid #CCC'}}>
            <div>Drop an image or click to select a file to upload.</div>
          </Dropzone>
        </div>

        <div className="imageDisplay col-md-10">
          {this.state.uploadedFile === null ? null :
          <div>
            <Editor
              image={this.state.uploadedFile.preview}
              width={250}
              height={250}
              borderRadius={0}
              scale={1}
              rotate={0}
            />
          </div>}
        </div>
      </form>
      </div>
      </div>
    )
  }
}

export default App;