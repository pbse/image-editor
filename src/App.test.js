import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount, render } from 'enzyme';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('render correctly', () => {
  const tree = renderer.create(
    <App />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render children', () => {
    const app = mount(
        <App>
          <div>Drop an image or click to select a file to upload.</div>
        </App>
    );
    expect(app.html()).toMatchSnapshot();
});

it('should render an input HTML element', () => {
    const app = mount(
        <App>
          <div>Drop an image or click to select a file to upload.</div>
        </App>
      );
    expect(app.find('input').length).toEqual(1);
});

it('renders dynamic props on the input element', () => {
    const component = mount(
    	<App inputProps={{ style: 'display: none' }} />
    );
    expect(component.find('input').html()).toContain('style="display: none');
});

it('should check for the value of input', () => {
      const app = mount(
        <App />
      );
      expect(app.render().find('input').attr('value')).toBeUndefined();
      expect(app.render().find('input').attr('value', 10)).not.toBeUndefined();
      app.simulate('click');
      expect(app.render().find('input').attr('value')).toBeUndefined();
});


