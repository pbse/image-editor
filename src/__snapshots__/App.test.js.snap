exports[`test render correctly 1`] = `
<div
  className="container">
  <div
    className="row">
    <form>
      <div
        className="FileUpload col-md-2">
        <div
          className=""
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={
            Object {
              "border": "1px solid #CCC",
              "height": "80px",
              "margin": "20px",
              "width": "160px",
            }
          }>
          <div>
            Drop an image or click to select a file to upload.
          </div>
          <input
            accept="image/*"
            multiple={false}
            onChange={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            type="file" />
        </div>
      </div>
      <div
        className="imageDisplay col-md-10" />
    </form>
  </div>
</div>
`;

exports[`test should render children 1`] = `"<div class=\"container\"><div class=\"row\"><form><div class=\"FileUpload col-md-2\"><div class=\"\" style=\"margin: 20px; width: 160px; height: 80px; border: 1px solid #ccc;\"><div>Drop an image or click to select a file to upload.</div><input type=\"file\" accept=\"image/*\" style=\"display: none;\"></div></div><div class=\"imageDisplay col-md-10\"></div></form></div></div>"`;
